src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Security.cs

```diff
@@ -1284,84 +1292,95 @@ public class C
     }
 }
 ";
-            Func<bool, Action<ModuleSymbol>> attributeValidator = isFromSource => (ModuleSymbol module) =>
-            {
-                var assembly = module.ContainingAssembly;
-                var ns = module.GlobalNamespace.GetMember<NamespaceSymbol>("N");
-                var namedType = ns.GetMember<NamedTypeSymbol>("C");
-                var type = (Cci.ITypeDefinition)namedType;
-                var method = (Cci.IMethodDefinition)namedType.GetMember("Goo");
 
-                if (isFromSource)
+            var compilation = CreateStandardCompilation(source, options: TestOptions.UnsafeReleaseDll);
+            CompileAndVerify(compilation, symbolValidator: module =>
```

Comment:
>CompileAndVerify [](start = 12, length = 16)

We should keep source validation and pass sourceSymbolValidator

Reply:
For this perticular test, `SourceAssemblySymbol.GetSecurityAttributes` and `NamedTypeSymbol.GetSecurityInformation` are the exact kind of internal APIs that we are trying to get away from using in tests. Are you proposing testing security attributes for source symbols in a different way?

---
In reply to: [150625800](https://github.com/dotnet/roslyn/pull/22896#discussion_r150625800) [](ancestors = 150625800)